x/curating/spec/README.md

```diff
@@ -0,0 +1,209 @@
+# Curating Module Specification
+
+## Abstract
+
+This module provides functionality to curate social media content and earn rewards in cryptocurrency via market-based mechanisms.
+
+## Concepts
+
+First, a post is created with a `post` transaction, taking a hash of the content, and a deposit. The deposit is returned after a curation window.
+
+Posts are curated with `upvote` transactions and `moderate` transactions.
+
+Upvotes are based on quadratic voting, and take number of votes as an argument. They also takes a deposit that is returned after the curation window. Deposits are mean't to keep the system honest. For example, governance may choose to slash the deposit of upvotes that are engaging in collusion or a Sybil attack.
+
+Moderate transactions are for signalling that a post contains spam or illegal content. Moderated posts immediately go to governance for adjudication. Curation is paused during this time. For posts deemed as inappropriate, upvoters get slashed and the moderator gets a reward. For posts not deemed as inappropriate, curation resumes while the moderator gets slashed.
+
+Only data for posts currently being curated are stored on-chain. Everything else is deleted after the curation window completes and rewards are paid out.
+
+## State
+
+### Parameters
+
+```go
+type Params struct {
+	CurationWindow       	 time.Duration
+	DefaultPostDeposit       sdk.Coin
+	DefaultUpvoteDepost  	 sdk.Coin
+	DefaultModerationDeposit sdk.Coin
+	DefaultVoteAmount        sdk.Coin // amount for 1 vote
+	MaxNumVotes              int // upper-limit on QV voting
+	RewardPoolAllocation 	 sdk.Dec // from inflation
+}
+```
+
+### Base Types
+
+```go
+type Post struct {
+	VendorID        uint64
+	PostID          uint64 // or string?
+	Creator         sdk.AccAddress
+	BodyHash        string
+	Deposit         sdk.Coin
+	CurationEndTime time.Time
+}
+```
+
+```go
+type Upvote struct {
+	VendorID        uint64
```

Comment:
Is `VendorID` really needed here if it's already part of the Post struct?

Reply:
It's technically not even needed here because it's part of the key. We may need to filter by vendor when calculating rewards. After I finish that part I'll revisit this to see if it's needed.